### ISSUE TITLE
Common time series dataset to facilitate model training and evaluation

### ISSUE THREAD

**Issue by ben-jy** [ontime-re/ontime#83]
When evaluating different models, datasets and their parameters may slightly differ. It would be beneficial to have a common dataset class that allows to define the important parameters for constructing a dataset, i.e. :

- the covariate components (if needed, past, future, static, etc.)
- the target components (components we must predict)
- the split (train-val-test)
- normalization
- ...

**Comment by ben-jy** [comment]
we already have a “BenchmarkDataset” class that seems to cover most of the points listed above. Why would we need another dataset class?

**Comment by ben-jy** [comment]
> we already have a “BenchmarkDataset” class that seems to cover most of the points listed above. Why would we need another dataset class?

Some model implementations, such as Moirai e.g., can receive specific parameters for their fit/predict methods that might improve quality of prediction, such as covariate, static features, etc.
We must find a way to give these parameters in a common way across models, that is why this issue was created. With a new time series inheriting from onTime TimeSeries object, we could give this information, and model could check on them.
Still, maybe we can find another way to perform this.

**Comment by ben-jy** [comment]
We might not need to change anything. 
1. When using a model independently from the benchmark module, we can simply add arguments to a method by creating a private method. E.g. create a private fit method in model wrapper to filter out keyword arguments given to the public fit method. This way, the method signature defined by the interface is not modified.
```
# create a private fit method that accepts a list of components that will be used as covariates
def _fit(self, ts: TimeSeries, covariates: Optional[List[str] = None, **params):
 # your implementation
 return

# public fit method
def fit(self, ts: TimeSeries, **params) -> "MyModel":
    return self._fit(ts, **params)

# usage
model = MyModel()
model.fit(ts, ["temp", "press"]) # we assume ts is defined
```

2. To be able to pass these parameters during benchmarking (where model is evaluated on many datasets), we should either : 
     a. add an attribute to the model wrapper, and pass it using the dynamic_model_params attribute from BenchmarkModelConfig class
  ```
  class MyModel(AbstractModel):
      def __init__(
          self,
          some_param,
          covariates: Optional[List[str]] = None,
      ):
          self.some_param = some_param
          self.covariates = covariates
  
      # public fit method
      def fit(self, ts: TimeSeries, **params) -> "MyModel":
          # use self.covariates somewhere in your implemnetion
          return
  
# usage
covariates_param = lambda ds: ["temp", "press"] if ds.name == "Meteo" else None
model_config = BenchmarkModelConfig(model_name="MyModel", model_class="MyModel", dynamic_model_params={"covariates": covariates_param})
# benchmark initialization
```
b. create a new attribute in BenchmarkModelConfig that resolves params for a specific model method (e.g. fit()). It would work exactly like dynamic_model_params, but would need to change Benchmark class to resolve them at the right time.


Solution b. seems the most appropriate.
